frontend/scripts/componentSchemas/types.ts

```diff
@@ -0,0 +1,36 @@
+import type { KeyValuePairs } from 'app-shared/types/KeyValuePairs';
+import type { ComponentType } from 'app-shared/types/ComponentType';
+import type { validVersions } from './version';
+
+export type AppFrontendVersion = (typeof validVersions)[number];
+
+export type ComponentName = keyof typeof ComponentType;
+
+export type PrefixedComponentName = `Comp${ComponentName}`;
+
+export interface LayoutSchema {
+  $schema: string;
+  $id: string;
+  $ref: string;
+  definitions: KeyValuePairs<KeyValuePairs>;
+}
+
+export type CondensedComponentSchema =
+  | {
+      title?: never;
+      anyOf?: never;
+      allOf: KeyValuePairs[];
+    }
+  | {
+      title: string;
+      anyOf: KeyValuePairs[];
+      allOf?: never;
+    };
```

Comment:
I see what you have tried here, but the `never` type is often used for things that never is supposed to happen, so then I'm wondering if we need it at all? 🤔 
Maybe we can just split the two scenarios into two types, something like this (with better names 😅): 
```suggestion
export type TypeA = {
   allOf: KeyValuePairs[];
}

export type TypeB = {
   title: string;
   anyOf: KeyValuePairs[];
}

export type CondensedComponentSchema = TypeA | TypeB;
```

By doing this we say that the `CondensedComponentSchema` can be either TypeA or TypeB 😄

Reply:
Updated 😊 
```ts
export type AllOfDefinition = {
  allOf: KeyValuePairs[];
};

export type AnyOfDefinition = {
  title: string;
  anyOf: KeyValuePairs[];
};

export type CondensedComponentSchema = AllOfDefinition | AnyOfDefinition;
````

Reply:
Great job @ErlingHauan ! 👏